src/lib/ContractTemplates/ERC721UserMintable.js

```diff
@@ -0,0 +1,942 @@
+import { ethers, utils } from 'ethers';
+import { TEMPLATES } from '../NFT/constants.js';
+import smartContractArtifact from './artifacts/ERC721UserMintable.js';
+import { isBoolean, isDefined } from '../utils.js';
+import { networkErrorHandler, errorLogger, ERROR_LOG } from '../error/handler.js';
+
+export default class ERC721UserMintable {
+  #gasLimit = 6000000;
+
+  contractAddress;
+
+  #contractDeployed;
+
+  #signer;
+
+  #template = TEMPLATES.ERC721UserMintable;
+
+  constructor(signer) {
+    this.#signer = signer;
+  }
+
+  getTemplate() {
+    return this.#template;
+  }
+
+  /* eslint-disable class-methods-use-this */
+  #addGasPriceToOptions(options, gas) {
+    const newOptions = options;
+    if (gas) {
+      if (typeof parseFloat(gas) !== 'number') {
+        throw new Error(
+          errorLogger({
+            location: ERROR_LOG.location.ERC721UserMintable_addGasPriceToOptions,
+            message: ERROR_LOG.message.invalid_gas_price_supplied,
+          }),
+        );
+      }
+      try {
+        const gasPrice = ethers.utils.parseUnits(gas, 'gwei');
+        newOptions.gasPrice = gasPrice;
+      } catch (error) {
+        const { message, type } = networkErrorHandler(error);
+        throw new Error(
+          errorLogger({
+            location: ERROR_LOG.location.ERC721UserMintable_addGasPriceToOptions,
+            message: ERROR_LOG.message.an_error_occured,
+            options: `${type} ${message}`,
+          }),
+        );
+      }
+    }
+    return newOptions;
+  }
+
+  /**
+   * Deploy ERC721UserMintable Contract. Used by the SDK class
+   * @param {string} name Name of the contract
+   * @param {string} symbol Symbol of the contract
+   * @param {string} baseURI baseURI for the contract
+   * @param {string} maxSupply Maximum supply of tokens for the contract
+   * @param {string} maxSupply Price to mint one token
+   * (link to a JSON file describing the contract's metadata)
+   * @returns void
+   */
+  async deploy({ name, symbol, baseURI, maxSupply, price, maxTokenRequest, gas = null }) {
+    if (this.contractAddress || this.#contractDeployed) {
+      throw new Error(
+        errorLogger({
+          location: ERROR_LOG.location.ERC721UserMintable_deploy,
+          message: ERROR_LOG.message.contract_already_deployed,
+        }),
+      );
+    }
+
+    if (!this.#signer) {
+      throw new Error(
+        errorLogger({
+          location: ERROR_LOG.location.ERC721UserMintable_deploy,
+          message: ERROR_LOG.message.no_signer_instance_supplied,
+        }),
+      );
+    }
+
+    if (!name) {
+      throw new Error(
+        errorLogger({
+          location: ERROR_LOG.location.ERC721UserMintable_deploy,
+          message: ERROR_LOG.message.no_name_supplied,
+        }),
+      );
+    }
+
+    if (symbol === undefined) {
```

Comment:
what if `symbol === null ` ??? i think it's better if you use `!symbol`, i know it's a copy past but are we allowed `null` as value for symbol ?